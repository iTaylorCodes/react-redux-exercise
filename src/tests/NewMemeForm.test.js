import { render, fireEvent } from "@testing-library/react";
import MemeForm from "../MemeForm";

let memeForm;

describe("<MemeForm /> basics", () => {
  beforeEach(() => {
    memeForm = render(<MemeForm />);
  });

  it("renders", () => {
    expect(memeForm).toBeTruthy();
  });
  it("matches snapshot", () => {
    expect(memeForm.asFragment()).toMatchSnapshot();
  });
});

it("updates state with topText input", () => {
  // mount form and mock addMeme
  const mockFn = jest.fn();
  const { getByText } = render(<MemeForm addMeme={mockFn} />);

  fireEvent.click(getByText("Create Meme!"));

  expect(mockFn).toHaveBeenCalled();
});
