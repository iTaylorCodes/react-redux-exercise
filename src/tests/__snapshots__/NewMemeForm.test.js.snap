// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MemeForm /> basics matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Make a Meme
    </h2>
    <form>
      <label
        for="url"
      >
        Image URL
      </label>
      <input
        id="form_url"
        name="url"
        type="text"
        value=""
      />
      <label
        for="topText"
      >
        Top Text
      </label>
      <input
        id="form_topText"
        name="topText"
        type="text"
        value=""
      />
      <label
        for="bottomText"
      >
        Bottom Text
      </label>
      <input
        id="form_bottomText"
        name="bottomText"
        type="text"
        value=""
      />
      <button
        id="form_submit"
        type="submit"
      >
        Create Meme!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
